processing/src/HpsEventFile.cxx

```diff
@@ -31,12 +31,13 @@ void HpsEventFile::close() {
 
 bool HpsEventFile::nextEvent() {
   
-  ++entry_;
-  if (entry_>maxEntries_)
+  if (entry_ > maxEntries_ - 1)
     return false;
 
   //TODO Really don't like having the tree associated to the event object. Should be associated to the EventFile.
   intree_->GetEntry(entry_);
+
+  ++entry_;
```

Comment:
Looking at this carefully, I agree that the ++entry_ should happen after GetEntry is called.

Reply:
The cleanest way to do this would be get rid of line 40 and call intree_->GetEntry(entry_++) on line 38

Reply:
Thanks for the comment. The change has been done.